tsdoc/src/parser/NodeParser.ts

```diff
@@ -559,22 +598,58 @@ export class NodeParser {
       closingDelimiterExcerpt: new Excerpt(closingDelimiterExcerptParameters)
     };
 
+    const tagNameWithUpperCase: string = tagName.toUpperCase();
+
     // Create a new TokenReader that will reparse the tokens corresponding to the tagContent.
     const embeddedTokenReader: TokenReader = new TokenReader(this._parserContext,
       tagContentExcerpt ? tagContentExcerpt.content : TokenSequence.createEmpty(this._parserContext));
 
-    switch (tagName.toUpperCase()) {
+    let docNode: DocNode;
+    switch (tagNameWithUpperCase) {
       case StandardTags.inheritDoc.tagNameWithUpperCase:
-        return this._parseInheritDocTag(docInlineTagParameters, embeddedTokenReader);
+        docNode = this._parseInheritDocTag(docInlineTagParameters, embeddedTokenReader);
+        break;
       case StandardTags.link.tagNameWithUpperCase:
-        return this._parseLinkTag(docInlineTagParameters, embeddedTokenReader);
+        docNode = this._parseLinkTag(docInlineTagParameters, embeddedTokenReader);
+        break;
       default:
-        return new DocInlineTag(docInlineTagParameters);
+        docNode = new DocInlineTag(docInlineTagParameters);
+    }
+
+    // Validate the tag
+    const tagDefinition: TSDocTagDefinition | undefined
+      = this._parserContext.configuration.tryGetTagDefinitionWithUpperCase(tagNameWithUpperCase);
+
+    if (tagDefinition) {
+      if (tagDefinition.syntaxKind !== TSDocTagSyntaxKind.InlineTag) {
+        // The tag is defined, but not as an inline tag
+        this._parserContext.log.addMessageForTokenSequence(
+          `The TSDoc tag "${tagName}" is not defined as an inline tag; it should not be enclosed in "{ }" braces`,
+          tagNameExcerptParameters.content, docNode);
+      } else {
+        if (!this._parserContext.configuration.validation.ignoreUnsupportedTags) {
+          if (!this._parserContext.configuration.isTagSupported(tagDefinition)) {
+            // The tag is defined, but not supported
+            this._parserContext.log.addMessageForTokenSequence(
+              `The TSDoc tag "${tagName}" is not supported by this tool`,
+              tagNameExcerptParameters.content, docNode);
+          }
+        }
+      }
+    } else {
+      // The tag is not defined
+      if (!this._parserContext.configuration.validation.ignoreUndefinedTags) {
+        this._parserContext.log.addMessageForTokenSequence(
+          `The TSDoc tag "${tagName}" is not defined in this configuration`,
+          tagNameExcerptParameters.content, docNode);
+      }
```

Comment:
Couldn't you extract this into a function shared between the tag kinds?

Reply:
Fixed

---
In reply to: [222150664](https://github.com/Microsoft/tsdoc/pull/80#discussion_r222150664) [](ancestors = 222150664)